tests/test_auth.py

```diff
@@ -0,0 +1,35 @@
+from httpx import Cookies
+
+from app import models
+from app.auth_manager import get_strategy
+from app.config import settings
+from app.utils.enums import RequestMethod
+from tests.utils import make_http_request
+
+
+async def test_refresh_token_handling(test_user: models.User):
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (testing):** Add tests for refresh token expiration and invalidation scenarios.

It's crucial to verify how the system behaves when the refresh token is expired or invalidated. Testing these scenarios ensures the security and reliability of the token refresh mechanism.

```suggestion
async def test_refresh_token_expired_handling(test_user: models.User):
    """
    Test function to verify the system's behavior when the refresh token is expired.

    Args:
        test_user: The test user object for token refresh testing.

    Returns:
        None
    """

    strategy = get_strategy()
    expired_token = await strategy.write_refresh_token(test_user, expire_in=-1)  # Assuming the strategy allows setting a negative expiration to simulate an expired token

    cookies = Cookies(
        {
            settings.refresh_token_name: expired_token,
        }
    )

    endpoint = "/api/users/me"

    res = await make_http_request(
        url=endpoint, method=RequestMethod.GET, cookies=cookies
    )
    assert res.status_code == 401  # Assuming 401 Unauthorized is the expected response for expired tokens


async def test_refresh_token_invalidated_handling(test_user: models.User):
    """
    Test function to verify the system's behavior when the refresh token is invalidated.

    Args:
        test_user: The test user object for token refresh testing.

    Returns:
        None
    """

    strategy = get_strategy()
    valid_token = await strategy.write_refresh_token(test_user)
    await strategy.invalidate_refresh_token(valid_token)  # Assuming the strategy has a method to invalidate tokens

    cookies = Cookies(
        {
            settings.refresh_token_name: valid_token,
        }
    )

    endpoint = "/api/users/me"

    res = await make_http_request(
        url=endpoint, method=RequestMethod.GET, cookies=cookies
    )
    assert res.status_code == 401  # Assuming 401 Unauthorized is the expected response for invalidated tokens
```